wp-content/themes/core/assets/js/src/admin/editor/components.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * @module
+ * @exports init
+ * @description Initializes theme component JS in the block editor context.
+ */
+
+import * as tests from 'utils/tests';
```

Comment:
If we remove `! tests.supportsWorkers()`, then we still need to set the wp variable importing wp-settings.js
```diff
- import * as tests from 'utils/tests';
+ import 'config/wp-settings';
```

Reply:
@sejas, both @dpellenwood and @Vinsanity mentioned that they needed to disable service workers in order to get this working in their implementations. I assumed that meant there is a service worker for the block editor that—among other things—provides caching, and that this must prevent some of the resources needed for custom component JS from being reachable.

I didn't look into it and just rolled with that. Perhaps @dpellenwood or @Vinsanity, you could lend some insight on this? In the meantime, I will dig in and see if I can get you a real answer on why this isn't compatible with the service works (or maybe it is!).